src/modules/edc-demo/components/asset-select/asset-select.component.ts

```diff
@@ -0,0 +1,47 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {MatDialog} from '@angular/material/dialog';
+import {filter, map} from 'rxjs/operators';
+import {Asset} from '../../models/asset';
+import {AssetDetailDialogData} from '../asset-detail-dialog/asset-detail-dialog-data';
+import {AssetDetailDialogResult} from '../asset-detail-dialog/asset-detail-dialog-result';
+import {AssetDetailDialog} from '../asset-detail-dialog/asset-detail-dialog.component';
+
+@Component({
+  selector: 'asset-select',
+  templateUrl: './asset-select.component.html',
+})
+export class AssetSelectComponent {
+  @Input()
+  label!: string;
+
+  @Input()
+  control!: FormControl<Asset[]>;
+
+  @Input()
+  assets: Asset[] = [];
+
+  @Output()
+  deleteDone = new EventEmitter();
```

Comment:
I don't think this is required. You correctly start the dialog with deletion disabled.